Release Ap-Nimbus v2.0.0

## Summary

### [appredict-docker](https://github.com/CardiacModelling/appredict-docker)

- [x] Update to [Chaste 2024.1](https://github.com/Chaste/Chaste/releases/tag/2024.1) and [ApPredict 2024.1](https://github.com/Chaste/ApPredict/releases/tag/v2024.1)
- [x] Upgrade base image to debian bullseye.
- [x] Build [appredict-chaste-libs](https://hub.docker.com/r/cardiacmodelling/appredict-chaste-libs):2.0.0 docker image.
- [x] Build [appredict-no-emulators](https://hub.docker.com/r/cardiacmodelling/appredict-no-emulators):2.0.0 docker image.
- [x] Build [appredict-with-emulators](https://hub.docker.com/r/cardiacmodelling/appredict-with-emulators):2.0.0 docker image.
- [x] Release repository v2.0.0

### [ap-nimbus-app-manager](https://github.com/CardiacModelling/ap-nimbus-app-manager)
- [x] Update to appredict-with-emulators:2.0.0 base image.
- [x] Upgrade to [Node 20](https://nodejs.org/en/blog/release/v20.15.0).
- [x] Build [ap-nimbus-app-manager](https://hub.docker.com/r/cardiacmodelling/ap-nimbus-app-manager):2.0.0 docker image.
- [x] Release repository v2.0.0

### [ap-nimbus-client](https://github.com/CardiacModelling/ap-nimbus-client)
- [x] Support Python 3.12
- [x] Build [ap-nimbus-client](https://hub.docker.com/r/cardiacmodelling/ap-nimbus-client/):2.0.0 docker image.
- [x] User Acceptance Testing
- [x] Release repository v2.0.0

### [ap-nimbus](https://github.com/CardiacModelling/ap-nimbus)
- [x] Update RtD documentation.
- [x] Update submodules for appredict-docker, app-manager, and ap-nimbus client to v2.0.0.
- [ ] Release repository v2.0.0

It looks like the numpy 2 errors [here](https://github.com/CardiacModelling/ap-nimbus-client/actions/runs/9681777837/job/26713158886) are pint-related (https://github.com/hgrecco/pint/issues/1974).

```
==================================== ERRORS ====================================
__________________ ERROR collecting files/tests/test_forms.py __________________
client/files/tests/test_forms.py:8: in <module>
    from files.forms import CellmlModelForm
client/files/forms.py:5: in <module>
    from cellmlmanip import load_model
/opt/hostedtoolcache/Python/3.10.14/x64/lib/python3.10/site-packages/cellmlmanip/__init__.py:3: in <module>
    from .main import load_model  # noqa
/opt/hostedtoolcache/Python/3.10.14/x64/lib/python3.10/site-packages/cellmlmanip/main.py:2: in <module>
    from cellmlmanip.parser import Parser
/opt/hostedtoolcache/Python/3.10.14/x64/lib/python3.10/site-packages/cellmlmanip/parser.py:15: in <module>
    from cellmlmanip.model import SYMPY_SYMBOL_DELIMITER, Model
/opt/hostedtoolcache/Python/3.10.14/x64/lib/python3.10/site-packages/cellmlmanip/model.py:15: in <module>
    from cellmlmanip.units import UnitStore
/opt/hostedtoolcache/Python/3.10.14/x64/lib/python3.10/site-packages/cellmlmanip/units.py:13: in <module>
    import pint
/opt/hostedtoolcache/Python/3.10.14/x64/lib/python3.10/site-packages/pint/__init__.py:28: in <module>
    from .measurement import Measurement
/opt/hostedtoolcache/Python/3.10.14/x64/lib/python3.10/site-packages/pint/measurement.py:12: in <module>
    from .quantity import Quantity
/opt/hostedtoolcache/Python/3.10.14/x64/lib/python3.10/site-packages/pint/quantity.py:69: in <module>
    from .numpy_func import (
/opt/hostedtoolcache/Python/3.10.14/x64/lib/python3.10/site-packages/pint/numpy_func.py:875: in <module>
    implement_single_dimensionless_argument_func(func_str)
/opt/hostedtoolcache/Python/3.10.14/x64/lib/python3.10/site-packages/pint/numpy_func.py:864: in implement_single_dimensionless_argument_func
    func = getattr(np, func_str)
/opt/hostedtoolcache/Python/3.10.14/x64/lib/python3.10/site-packages/numpy/__init__.py:410: in __getattr__
    raise AttributeError("module {!r} has no attribute "
E   AttributeError: module 'numpy' has no attribute 'cumproduct'. Did you mean: 'cumprod'?
```

--- a/.github/workflows/docker-no-emulators.yml
+++ b/.github/workflows/docker-no-emulators.yml
@@ -29,6 +29,10 @@ jobs:
           context: ./appredict-no-emulators/
           pull: true
           push: true
+          build-args: |
+            appredict_tag=v2024.1
+            chaste_tag=2024.1
+            build_processors=4
           tags: |
             cardiacmodelling/appredict-no-emulators:2.0.0
             cardiacmodelling/appredict-no-emulators:latest

--- a/.github/workflows/docker-with-emulators.yml
+++ b/.github/workflows/docker-with-emulators.yml
@@ -29,6 +29,8 @@ jobs:
           context: ./appredict-with-emulators/
           pull: true
           push: true
+          build-args: |
+            build_processors=4
           tags: |
             cardiacmodelling/appredict-with-emulators:2.0.0
             cardiacmodelling/appredict-with-emulators:latest

--- a/LICENSE
+++ b/LICENSE
@@ -1,7 +1,7 @@
 BSD 3-Clause License
 
 Unless otherwise noted, all files within this distribution are
-Copyright (c) 2013-2021 University of Oxford, 2016-2021 University College London, and 2016-2021 University of Nottingham.
+Copyright (c) 2013-2021 University of Oxford, 2016-2021 University College London, and 2016-2024 University of Nottingham.
 All rights reserved.
 
 University of Oxford means the Chancellor, Masters and Scholars of the

--- a/appredict-chaste-libs/Dockerfile
+++ b/appredict-chaste-libs/Dockerfile
@@ -1,13 +1,13 @@
-FROM debian:buster-slim
+FROM debian:bullseye-slim
 
 USER root
 
-RUN echo "deb http://deb.debian.org/debian buster main contrib non-free" >> /etc/apt/sources.list
+RUN echo "deb http://deb.debian.org/debian bullseye main contrib non-free" >> /etc/apt/sources.list
 
 # Install dependencies with recommended, applicable suggested and other useful packages
 RUN apt-get update && \
-    apt-get install -y g++ cmake && \ 
-    apt-get install -y --no-install-recommends \ 
+    apt-get install -y g++ cmake && \
+    apt-get install -y --no-install-recommends \
     wget \
     git \
     petsc-dev \

--- a/appredict-no-emulators/Dockerfile
+++ b/appredict-no-emulators/Dockerfile
@@ -4,8 +4,8 @@ FROM cardiacmodelling/appredict-chaste-libs:1.0.0
 # Modifiable (e.g. via command line) args.                                     #
 ################################################################################
 
-ARG appredict_tag=v2021.1
-ARG chaste_tag=2021.1
+ARG appredict_tag=v2024.1
+ARG chaste_tag=2024.1
 ARG build_processors=1
 
 ################################################################################

--- a/appredict-no-emulators/Dockerfile
+++ b/appredict-no-emulators/Dockerfile
@@ -37,7 +37,7 @@ RUN find . ! -name shannon_wang_puglisi_weber_bers_2004.cellml ! -name ten_tussc
 
 RUN mkdir -p ${dir_build}
 WORKDIR ${dir_build}
-RUN cmake -j ${build_processors} \
+RUN cmake \
     -DCMAKE_BUILD_TYPE=Release \
     -DChaste_ERROR_ON_WARNING=OFF \
     -DChaste_UPDATE_PROVENANCE=ON \

--- a/appredict-no-emulators/Dockerfile
+++ b/appredict-no-emulators/Dockerfile
@@ -1,4 +1,4 @@
-FROM cardiacmodelling/appredict-chaste-libs:1.0.0
+FROM cardiacmodelling/appredict-chaste-libs:2.0.0
 
 ################################################################################
 # Modifiable (e.g. via command line) args.                                     #

--- a/appredict-with-emulators/Dockerfile
+++ b/appredict-with-emulators/Dockerfile
@@ -1,4 +1,4 @@
-FROM cardiacmodelling/appredict-no-emulators:1.0.0
+FROM cardiacmodelling/appredict-no-emulators:2.0.0
 
 ################################################################################
 # Modifiable (e.g. via command line) args.                                     #

